front/lib/api/assistant/messages.ts

```diff
@@ -57,6 +64,80 @@ export function getMaximalVersionAgentStepContent(
   return Array.from(maxVersionStepContents.values());
 }
 
+export async function generateParsedContents(
```

Comment:
can we make this a pure function that takes the step contents ? Pretty sure we already have them at call site.

Reply:
we don't if I remove the include AgentStepContents in the message fetch

Reply:
i think

Reply:
https://github.com/dust-tt/dust/pull/14901/files/aa8893c71f9b91e7e08e14ca1d174bb191362357#diff-e2d9cb072451a0ccbf568fd15477f8bcffd696dd9b6ac474288465b9062fdb8dR358

line 358